docs/hub/spaces.md

```diff
@@ -169,8 +169,28 @@ jobs:
 
 ## Troubleshoot
 
+### Streamlit
+
 Issues may occur when you use an unsupported Streamlit version. The Streamlit version is not configured in the **requirements.txt** file but rather in the `YAML` settings through the `sdk_version` setting. Not all Streamlit versions are supported. Check that you are using a supported version of Streamlit. Refer to the [reference section](#reference) for more information about supported versions.
 
+### Custom Python Spaces
+
+While not an official workflow, you are able to run your own Python + interface stack in Spaces by selecting Gradio as your SDK and serving a frontend on port `7680`. See the [templates](https://huggingface.co/templates#spaces) for examples.
+
+Spaces are served in iframes, which by default restrict links from opening in the parent page. The simplest solution is to open them a new window:
+
+```HTML
+<a href="https://hf.space" rel="noopener" target="_blank">Spaces</a>
+```
+
+Usually, the height of Spaces is adjusted automatically when using the Gradio library interface. If you provide your own frontend in the Gradio SDK however, you'll need to add an [iFrame Resizer script](https://cdnjs.com/libraries/iframe-resizer) so that content can scroll in the iframe:
```

Comment:
```suggestion
Usually, the height of Spaces is adjusted automatically when using the Gradio library interface. If you provide your own frontend in the Gradio SDK however, and if your content height is larger than the viewport, you'll need to add an [iFrame Resizer script](https://cdnjs.com/libraries/iframe-resizer) so that content can scroll in the iframe:
```